### PR TITLE
Limit interaction

### PR DESCRIPTION
**Pull request by bbenligiray**
Closes https://github.com/api3dao/claims-manager/issues/11 and https://github.com/api3dao/claims-manager/issues/10

### CODE REVIEW

**Review by ErichDylus — APPROVED** [review]
Code looks good to me. The only nits I would change in the comments, I guess more for documentation purposes if it matters here, are:
(1) option 0 of DoNotPay could also be arbitrator rejecting payment of both the claimed amount and the settlement amount based on the evidence (rather than refusing to arbitrate entirely), and 
(2) changing "appeal to this" to "appeal this" to be clear it's being appealed rather than the response to an appeal

**Review by acenolaza — APPROVED** [review]

File: contracts/ClaimsManagerWithKlerosArbitration.sol [hunk 1]
```diff
@@ -148,6 +148,31 @@ contract ClaimsManagerWithKlerosArbitration is
...
+            IAccessControlRegistry(accessControlRegistry).hasRole(
+                mediatorRole,
+                msg.sender
+            )
+        ) {
+            require(
+                arbitrator.currentRuling(disputeId) != 2,
+                "Ruling agrees with mediator"
+            );
+        } else {
```
Comment (acenolaza) [inline contracts/ClaimsManagerWithKlerosArbitration.sol:173]: This is just a my personal preference but I find this easier to read:

```suggestion
        if (arbitrator.currentRuling(disputeId) == 1) {
            require(manager == msg.sender ||
            IAccessControlRegistry(accessControlRegistry).hasRole(
                mediatorRole,
                msg.sender
            ), "Only mediator can appeal pay claim" );
        } else if (arbitrator.currentRuling(disputeId) == 2) {
            require(msg.sender == claims[claimIndex].claimant,
                "Only claimant can appeal pay settlement"
            );
        } else {
```

Comment (bbenligiray) [reply contracts/ClaimsManagerWithKlerosArbitration.sol:173]: Kinda difficult to implement the else though

Comment (acenolaza) [reply contracts/ClaimsManagerWithKlerosArbitration.sol:173]: You are right. Didn't think about the `else` block 🤦🏻  but maybe we can get rid of it and just create a modifier `onlyAppealer` where we check that `msg.sender` is either claimant or manager/mediator. 

Again, I believe this suggestion makes the code easier to read but maybe I'm wrong. It's your call

Comment (bbenligiray) [reply contracts/ClaimsManagerWithKlerosArbitration.sol:173]: I'll keep it as is because it's more gas efficient this way

**Pull request merged**